x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/CurrentDate.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.expression.function.scalar.datetime;
+
+import org.elasticsearch.xpack.sql.expression.function.scalar.ConfigurationFunction;
+import org.elasticsearch.xpack.sql.session.Configuration;
+import org.elasticsearch.xpack.sql.tree.NodeInfo;
+import org.elasticsearch.xpack.sql.tree.Source;
+import org.elasticsearch.xpack.sql.type.DataType;
+import org.elasticsearch.xpack.sql.util.DateUtils;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+
+public class CurrentDate extends ConfigurationFunction {
+
+    private final ZonedDateTime date;
+
+    public CurrentDate(Source source, Configuration configuration) {
```

Comment:
Instead of providing a new function, the logic from `CurrentDateTime` should be reused - I would refactor that a package protected `CurrentFunction` and have `CurrentDate`/`Timestamp` extensions that would just trim the precision in the delegating constructor as well as specifying the dedicated type.

Reply:
I don't see a nice way to do it unless the `date` attribute in the new abstract class is not final and is not set in the constructor.

Reply:
You cannot do a method call in the `super()`, it's not allowed.

Reply:
You can if the method is static - a constructor is simply a special method. One is not allowed to call a (virtual) method from the about to be created class: 
`super(source, DateUtils.asDateOnly(configuration.now()), DataType.DATE);`